pkg/cmd/util/tokencmd/request_token.go

```diff
@@ -184,32 +204,34 @@ func (o *RequestTokenOptions) RequestToken() (string, error) {
 	}
 }
 
-func oauthAuthorizeResult(location string) (string, error) {
-	u, err := url.Parse(location)
+func oauthAuthorizeResult(client *osincli.Client, authorizeRequest *osincli.AuthorizeRequest, location string) (string, error) {
+	// Make a request out of the URL since that is what AuthorizeRequest.HandleRequest expects to extra data from
+	req, err := http.NewRequest("GET", location, nil)
 	if err != nil {
 		return "", err
 	}
 
-	if errorCode := u.Query().Get("error"); len(errorCode) > 0 {
-		errorDescription := u.Query().Get("error_description")
-		return "", errors.New(errorCode + " " + errorDescription)
+	authorizeData, err := authorizeRequest.HandleRequest(req)
+	if err != nil {
+		return "", errIfOAuthError(err)
 	}
 
-	// Grab the raw fragment ourselves, since the stdlib URL parsing decodes parts of it
-	fragment := ""
-	if parts := strings.SplitN(location, "#", 2); len(parts) == 2 {
-		fragment = parts[1]
-	}
-	fragmentValues, err := url.ParseQuery(fragment)
+	accessRequest := client.NewAccessRequest(osincli.AUTHORIZATION_CODE, authorizeData)
+	accessData, err := accessRequest.GetToken()
 	if err != nil {
-		return "", err
+		return "", errIfOAuthError(err)
 	}
 
-	if accessToken := fragmentValues.Get("access_token"); len(accessToken) > 0 {
-		return accessToken, nil
-	}
+	return accessData.AccessToken, nil
+}
 
-	return "", nil
+func errIfOAuthError(err error) error {
```

Comment:
can you add a comment on why it's ok to drop other errors?  Superficially this seems like a change in behavior from the old code